Change default value of PodToolExecutionViaShellEnabled to be compatible with CocoaPods 11.

AlmostMatt opened the pull request:
Change default value of PodToolExecutionViaShellEnabled from true to false.
When shell execution is disabled, environment variables can be set explicitly for the cocoapods process instead of relying on the user's bash profile. This guarantees that LANG is set to a value that includes utf8, which is necessary when using CocoaPods version 1.11 or higher.

This resolves https://github.com/firebase/quickstart-unity/issues/1149

chkuang-g commented:
I made some dig about why this is set to `true` by default for IOSResolver, but remained `false` for AndroidResolver. And this commit is what I found.

https://github.com/googlesamples/unity-jar-resolver/commit/a77baf49acaca38d712f14278297b58387057cc5

Though this is 5 years old, there seems to be a need to customize shell environment, ex. for ssh certs, back then.  I think reverting this change can be a significant change in behavior for some users.

As we found out that `LANG=en_US.UTF-8` requirement only started from Cocoapod `1.11.x`, and we have no control about how long this requirement will last, for the convenience of our user, we should do the following:

- As you suggested, we can add `export LANG=en_US.UTF-8;` before the `pod` command [here](https://github.com/googlesamples/unity-jar-resolver/blob/master/source/IOSResolver/src/IOSResolver.cs#L2464). That is, change the command from `pod` to `export LANG=en_US.UTF-8; pod`.  This patch should only apply when the current platform is Mac.  You can use [Application.platform](https://docs.unity3d.com/ScriptReference/Application-platform.html) to determine the current platform.
- I think it is worth it to add a setting for this workaround since we have no idea how long this would be a requirement. The setting would be something like "Force set LANG=en_US.UTF-8 for pod commands on Mac". You can follow the pattern `PodToolExecutionViaShellEnabled` uses.

chkuang-g reviewed (changes requested):
I made some dig about why this is set to `true` by default for IOSResolver, but remained `false` for AndroidResolver. And this commit is what I found.

https://github.com/googlesamples/unity-jar-resolver/commit/a77baf49acaca38d712f14278297b58387057cc5

Though this is 5 years old, there seems to be a need to customize shell environment, ex. for ssh certs, back then.  I think reverting this change can be a significant change in behavior for some users.

As we found out that `LANG=en_US.UTF-8` requirement only started from Cocoapod `1.11.x`, and we have no control about how long this requirement will last, for the convenience of our user, we should do the following:

- As you suggested, we can add `export LANG=en_US.UTF-8;` before the `pod` command [here](https://github.com/googlesamples/unity-jar-resolver/blob/master/source/IOSResolver/src/IOSResolver.cs#L2464). That is, change the command from `pod` to `export LANG=en_US.UTF-8; pod`.  This patch should only apply when the current platform is Mac.  You can use [Application.platform](https://docs.unity3d.com/ScriptReference/Application-platform.html) to determine the current platform.
- I think it is worth it to add a setting for this workaround since we have no idea how long this would be a requirement. The setting would be something like "Force set LANG=en_US.UTF-8 for pod commands on Mac". You can follow the pattern `PodToolExecutionViaShellEnabled` uses.

AlmostMatt commented:
Ok, I changed the default value of PodToolExecutionViaShellEnabled back to true. I added a new setting called PodToolShellExecutionSetLang, which causes "export LANG=...; " to be added to the start of the arguments when shell execution is enabled for ios builds.

This requires bash as the tool instead of pod as the tool, which is technically conditional on stdoutRedirectionInShellMode, but that parameter is always true.

AlmostMatt commented on source/AndroidResolver/src/CommandLine.cs:
```diff
@@ -541,7 +541,7 @@ public void AggregateLine(Process process, StreamWriter stdin, StreamData data)
                 Dictionary<string, string> envVars = null,
                 IOHandler ioHandler = null, bool useShellExecution = false,
-                bool stdoutRedirectionInShellMode = true) {
+                bool stdoutRedirectionInShellMode = true, bool setLangInShellMode = false) {
```
Done.
        /// <param name="setLangInShellMode">Causes environment variable LANG to be explicitly set
        /// when  executing a command via the shell.  This is only applicable to OSX.</param>
